internal/replacements/replacements.go

```diff
@@ -95,7 +96,7 @@ func ProcessGroupNeedsRemoval(ctx context.Context, podManager podmanager.PodLife
 		return false, podErr
 	}
 
-	return processGroupNeedsRemovalForPod(cluster, pod, processGroup, log)
+	return processGroupNeedsRemovalForPod(cluster, pod, processGroup, log, hasPVC && replaceOnSecurityContextChange)
```

Comment:
Why are we checking if the `hasPVC` is true? What is the implication of not doing that?

Reply:
oh, I just figured if it doesn't have a PVC then it doesn't have to worry about file security context changes

Reply:
though this does not seem to work, when I tested with General process class it replaces stateless pods 🤔

Reply:
discussed offline, hasPVC doesnt really matter as we'd eventually replace for spec change anyways in defaults, and having stateless pods with matching specs is desirable